src/oscar/apps/basket/forms.py

```diff
@@ -61,13 +61,19 @@ def check_max_allowed_quantity(self, qty):
         # add second time, just updating number of items.
         qty_delta = qty - self.instance.quantity
         is_allowed, reason = self.instance.basket.is_quantity_allowed(qty_delta)
+        if self.instance.basket.strategy.request and self.instance.basket.strategy.request.is_ajax:
+            if self.instance.basket.strategy.request.POST.get('form-0-DELETE'):
```

Comment:
This also assumes that this form is the first item in the formset - there is no guarantee that this is the case.